tests/pester-tests/commands/choco-source.Tests.ps1

```diff
@@ -425,6 +542,61 @@ Describe "choco <_>" -ForEach $Command -Tag Chocolatey, SourceCommand {
         It "Displays no change made" {
             $Output.Lines | Should -Contain "Nothing to change. Config already set."
         }
+
+        Context "when using enhanced exit codes" {
+            BeforeAll {
+                $null = Enable-ChocolateyFeature -Name "useEnhancedExitCodes"
+
+                $Output = Invoke-Choco $CurrentCommand enable --name "not-existing"
+            }
+
+            It "Exits with ExitCode 2" {
+                $Output.ExitCode | Should -Be 2 -Because $Output.String
+            }
+
+            It "Changes Nothing" {
+                $Output.Lines | Should -Contain "Nothing to change. Config already set."
+            }
+        }
+    }
+
+    Context "Enabling source that is already enabled" {
+        BeforeAll {
+            $null = Disable-ChocolateyFeature -Name "useEnhancedExitCodes"
+
+            # Ensure source is enable
+            Invoke-Choco $CurrentCommand add --name "already-enabled" --source "https://somewhere/out/there/"

```

Comment:
It's not likely to be there already, but just in case somehow it's there, let's make sure it's enabled (and this makes it the same as the already-disabled test).

```suggestion
            Invoke-Choco $CurrentCommand add --name "already-enabled" --source "https://somewhere/out/there/"
            Invoke-Choco $CurrentCommand enable --name "already-enabled"
```